### PR TITLE
To include process arguments in user section event

### PR DESCRIPTION
**Pull request by hnadiminti-equinix**
Desc:
This PR is to include process arguments in user section event
- UserAction event occurs when a user executes a program
- Added unit test cases.
[SCT-305](https://packet.atlassian.net/browse/SCT-305)

### CODE REVIEW

File: internal/auditd/sessiontracker/sessiontracker.go [hunk 1]
```diff
@@ -324,11 +327,12 @@ func (o *sessionTracker) DeleteRemoteUserLoginsBefore(t time.Time) {
...
-	srcPID int
-	hasRUL bool
-	login  common.RemoteUserLogin
-	cached []*aucoalesce.Event
+	added    time.Time
+	srcPID   int
+	hasRUL   bool
+	login    common.RemoteUserLogin
+	cached   []*aucoalesce.Event
+	procArgs []string
```
Comment (stephen-fox) [inline internal/auditd/sessiontracker/sessiontracker.go:335]: You can remove this field - there is no need to store process arguments in this struct.

File: internal/auditd/sessiontracker/sessiontracker.go [hunk 2]
```diff
@@ -235,9 +235,12 @@ func (o *sessionTracker) auditEventWithoutSession(event *aucoalesce.Event, debug
...
 	}
 
+	procArgs := event.Process.Args
+
 	u := &user{
-		added:  time.Now(),
-		srcPID: srcPID,
+		added:    time.Now(),
+		srcPID:   srcPID,
+		procArgs: procArgs,
```
Comment (stephen-fox) [inline internal/auditd/sessiontracker/sessiontracker.go:243]: Per the comment below, these code changes can be removed.

File: internal/auditd/sessiontracker/sessiontracker.go [hunk 3]
```diff
@@ -364,6 +368,12 @@ func (o *user) toAuditEvent(ae *aucoalesce.Event) *auditevent.AuditEvent {
 		"auditd",
 	).WithTarget(o.login.Source.Target)
 
+	// adding process args
+	if evt.Metadata.Extra == nil {
+		evt.Metadata.Extra = make(map[string]any)
+	}
+	evt.Metadata.Extra["process_args"] = ae.Process.Args
```
Comment (stephen-fox) [inline internal/auditd/sessiontracker/sessiontracker.go:375]: Can you add a check to see if `ae.Process.Args` has a greater-than-zero length? That way we will only add a `process_args` map entry if process arguments are present.

E.g.,

```go
if len(ae.Process.Args) > 0 {
    evt.Metadata.Extra["process_args"] = ae.Process.Args
}
```

File: go.mod [hunk 4]
```diff
@@ -30,6 +30,7 @@ require (
 	github.com/prometheus/client_model v0.3.0 // indirect
 	github.com/prometheus/common v0.42.0 // indirect
 	github.com/prometheus/procfs v0.9.0 // indirect
+	github.com/rogpeppe/go-internal v1.10.0 // indirect
```
Comment (stephen-fox) [inline go.mod:33]: Why was this library added as a dependency?

Can you run `go mod tidy` and see if that modifies `go.mod`. If it remains the same, then no additional commit is required.

**Review by stephen-fox — CHANGES_REQUESTED** [review]

File: internal/auditd/sessiontracker/sessiontracker.go (hunk 2 above)
Comment (hnadiminti-equinix) [reply internal/auditd/sessiontracker/sessiontracker.go]: Updated as per the review...Thanks @stephen-fox

File: internal/auditd/sessiontracker/sessiontracker.go (hunk 3 above)
Comment (hnadiminti-equinix) [reply internal/auditd/sessiontracker/sessiontracker.go]: Updated

File: go.mod (hunk 4 above)
Comment (hnadiminti-equinix) [reply go.mod:33]: Fyi Please find the below screenshot :
image

**Comment by JAORMX** [comment]
Let's get some testing for this.

File: internal/auditd/sessiontracker/sessiontracker.go (hunk 1 above)
Comment (hnadiminti-equinix) [reply internal/auditd/sessiontracker/sessiontracker.go]: done

**Comment by hnadiminti-equinix** [comment]
> Let's get some testing for this.

Added the test cases , Please have a look and let me know the feedback.

**Review by stephen-fox — APPROVED** [review]

**Pull request merged**